### PR TITLE
Issues: 300, 288, 286, 269, 245, 198, 181, 312. 307

### PR DESCRIPTION
**Pull request by AmeNote-Michael**
This pull request covers multiple issues related to driver as listed below:

Issue #300: MidiSrv gets stuck by sending MT3.
- issue specific in sending empty SYSEX message "F0 F7", which was not being handled by driver. Driver now handles empty SYSEX.

Issue #288: Return manufacturer name from ACX driver

Issue #286: USB MIDI 2 Driver returns incorrect size for iSerialNumber property

Issue #269: MIDI.exe returns different GTB than returned via USB
- driver returned all GTBs defined - now returns only GTBs defined and Indexed by Endpoints.

Issue #245: MIDI Service gets stuck.
- earlier driver version tried to filter out undefined messages but got into an endless loop. Driver now will pass message on for upper layers to handle.

Issue #198: Roland-reported WDF_VIOLATION crash with MIDI 2.0 driver and Roland UM-One
- resolved in fixing power management issue. BE SURE you have UM-One that is compatible with Windows 11 and that you are in TAB mode for class compliance on the UM-One. There are some UM-Ones that state they are not compatible with Windows 11.

Issue #181: Green screen appears when assigned USB MIDI 2.0 driver to a UMP device
- resolved in fixing power management issue.

Issue #165: Green screen appears when assigned USB MIDI 2.0 driver to a UMP device
- Issue was due to architecture of USB Out packets from UMP. UMP presented from service layer as individual UMP packets (max 4 bytes). Driver now will try to consolidate into single max size USB packets if it knows more data available from Service.

Issue #312: UMP packet split across USB transaction is corrupted
- Issue was solved. Code did not manage properly splitting across multiple USB Transactions. With fix, running test code on ProtoZOA and procedure provided gives expected output.

Issue #307: Sending messages without monitoring causes Green Screen with the USB MIDI 2.0 driver
- Issue seems to have been resolved by fixing other issues listed. As per issue, ran testing and repeated the 125000 messages over 30 times without monitor running - no crash experienced.

Arm64 Driver Version ready for extensive testing. This version of driver is ready to be compiled and tested on Arm64 based Windows environments.

### CODE REVIEW

**Comment by AmeNote-Michael** [comment]
> @AmeNote-Michael do you want me to reject this PR given what you are currently working on, or do you intend to issue another PR afterwards with just the delta?

Lets complete the PR. There was a lot of changes and I would prefer to lock down here before I really dig into the changes for SpinLocks over Mutex's.

Wait for Attestation signing for next PR.

**Comment by Psychlist1972** [comment]
Please resolve the conflicts first. Also, attestation signing requires a fair bit of work from the Microsoft team, so prefer to hold on that until you have a complete version of the driver with the mutex changes.

**Comment by AmeNote-Michael** [comment]
@Psychlist1972 I have resolved conflict - missed removing one element of vcpkg addition.

File: src/api/Drivers/USBMIDI2/Driver/Device.cpp [hunk 1]
```diff
@@ -2280,7 +2344,7 @@ Return Value:Amy
         }
 
         // Check if anything leftover to write to USB
-        if (writeBufferIndex)
+        if (writeBufferIndex && !isMoreData) // and there is not more data coming
         {
             // NEED TO FILL REST OF BUFFER WITH NULL
```
Comment (Psychlist1972) [inline src/api/Drivers/USBMIDI2/Driver/Device.cpp:2349]: Is this step a to-do? (filling the rest of the buffer with null)

File: src/api/Drivers/USBMIDI2/Driver/USBMidi2.vcxproj [hunk 2]
```diff
@@ -111,6 +116,7 @@
     </DriverSign>
     <Link>
       <AdditionalDependencies>%(AdditionalDependencies);$(KernelBufferOverflowLib);$(DDK_LIB_PATH)ntoskrnl.lib;$(DDK_LIB_PATH)hal.lib;$(DDK_LIB_PATH)wmilib.lib;acx\km\$(ACX_VERSION_MAJOR).$(ACX_VERSION_MINOR)\acxstub.lib;$(DDK_LIB_PATH)\libcntpr.lib;$(DDK_LIB_PATH)\wpprecorder.lib;$(DDK_LIB_PATH)\Usbd.lib</AdditionalDependencies>
+      <TreatLinkerWarningAsErrors>false</TreatLinkerWarningAsErrors>
```
Comment (Psychlist1972) [inline src/api/Drivers/USBMIDI2/Driver/USBMidi2.vcxproj:119]: What linker warnings were you getting? This setting won't apply when we do the internal build.

File: src/api/Drivers/USBMIDI2/Driver/Device.cpp (hunk 1 above)
Comment (AmeNote-Michael) [reply src/api/Drivers/USBMIDI2/Driver/Device.cpp:2349]: This was a left over comment. Removed

File: src/api/Drivers/USBMIDI2/Driver/USBMidi2.vcxproj (hunk 2 above)
Comment (AmeNote-Michael) [reply src/api/Drivers/USBMIDI2/Driver/USBMidi2.vcxproj:119]: This was accidentally left in while trying to use vcpkg. Corrected.

**Comment by AmeNote-Michael** [comment]
@Psychlist1972 @garydan42 I believe this PR is ready for review, merging and Attestation signing. Please proceed.

**Review by Psychlist1972 — APPROVED** [review]

**Pull request merged**